core/sdk-core/src/main/java/software/amazon/awssdk/core/client/config/ClientOverrideConfiguration.java

```diff
@@ -582,25 +669,33 @@ default Builder compressionConfiguration(Consumer<CompressionConfiguration.Build
     /**
      * An SDK-internal implementation of {@link ClientOverrideConfiguration.Builder}.
      */
-    private static final class DefaultClientOverrideConfigurationBuilder implements Builder {
-        private Map<String, List<String>> headers = new HashMap<>();
-        private RetryPolicy retryPolicy;
-        private List<ExecutionInterceptor> executionInterceptors = new ArrayList<>();
-        private AttributeMap.Builder advancedOptions = AttributeMap.builder();
-        private Duration apiCallTimeout;
-        private Duration apiCallAttemptTimeout;
-        private Supplier<ProfileFile> defaultProfileFileSupplier;
-        private ProfileFile defaultProfileFile;
-        private String defaultProfileName;
-        private List<MetricPublisher> metricPublishers = new ArrayList<>();
-        private ExecutionAttributes.Builder executionAttributes = ExecutionAttributes.builder();
-        private ScheduledExecutorService scheduledExecutorService;
-        private CompressionConfiguration compressionConfiguration;
+    @SdkInternalApi
+    static final class DefaultBuilder implements Builder {
+        private final SdkClientConfiguration.Builder config;
+        private final SdkClientConfiguration.Builder resolvedConfig;
+
+        @SdkInternalApi
+        DefaultBuilder(SdkClientConfiguration.Builder config) {
+            this();
+            RESOLVED_OPTIONS.forEach(o -> copyValue(o, config, this.resolvedConfig));
+            CLIENT_OVERRIDE_OPTIONS.forEach(o -> copyValue(o, config, this.config));
+            SdkAdvancedClientOption.options().forEach(o -> copyValue(o, config, this.config));
```

Comment:
I don't think so, because `SdkAdvancedAsyncClientOption`s can't be stored in ClientOverrideConfiguration. They don't extend `SdkAdvancedClientOption`.